src/Components/Components/src/Routing/RouteTableFactory.cs

```diff
@@ -334,4 +298,70 @@ internal static int RouteComparison(InboundRouteEntry x, InboundRouteEntry y)
             _ => throw new InvalidOperationException("Invalid comparison result."),
         };
     }
+
+    private class InboundRouteEntryEqualityComparer : IEqualityComparer<InboundRouteEntry>
```

Comment:
Seal and use a singleton?
```suggestion
    private sealed class InboundRouteEntryEqualityComparer : IEqualityComparer<InboundRouteEntry>
```

Reply:
`sealed` sure but why use singleton if it will only ever be run once per app as far as I can tell?